### PR TITLE
x264: reduce size of 'full' build

### PR DESCRIPTION
**Pull request by myfreeer**
This commit disables non-video codecs, bsfs, all protocols, network support and doc of lightffmpeg to reduce size.

### CODE REVIEW

**Comment by wiiaboo** [comment]
What's the size difference? I certainly won't merge that huge fucking line. You can send --enable-decoder=* with the same name several times.

**Comment by myfreeer** [comment]
> What's the size difference?
15+MB before and  12.4MB after

**Comment by myfreeer** [comment]
> What's the size difference?

15+MB before and  12.4MB after

**Comment by wiiaboo** [comment]
3MB isn't that that much of a savings.

**Comment by wiiaboo** [comment]
If you really really want this, I'd suggest parsing libavcodec/allcodecs.c to extract the list of audio decoders (don't bother with external libs, since they're already disabled by default) to disable instead of having a hardcoded list on the script. And placing this option in a separate x264 option in the .bat with how much space it saves.

**Comment by wiiaboo** [comment]
This command run inside ffmpeg-git gives a comma-separated list of audio, DPCM and ADPCM decoders:
```bash
sed -n '/audio codecs/,/subtitles/p' libavcodec/allcodecs.c | \
grep -P 'REGISTER_(ENCDEC|DECODER)' | \
grep -oP "[a-z0-9_]+(?=\);)" | tr '\n' ',' | head -c -1
```

**Comment by wiiaboo** [comment]
Have you tried disabling all codecs and just explicitely enabling the ones you want?
That should be a much bigger size difference.

**Comment by myfreeer** [comment]
11.5MB for 64bit with commands below, not sure if some of them being unneeded...
```sh
decoders="012v,apng,ayuv,bmp,cfhd,cinepak,dds,dirac,dnxhd,\
ffv1,flashsv,flashsv2,flv,gif,h261,h263,h264,hevc,indeo2,indeo3,indeo4,\
indeo5,jpeg2000,mjpeg,mpeg1video,mpeg2video,mpegvideo,mpeg4,\
msmpeg4v1,msmpeg4v2,msmpeg4,mss1,mss2,msvideo1,pcx,png,prores,\
psd,qdraw,qtrle,r210,rawvideo,rpza,rv10,rv20,rv30,rv40,sgi,smc,svq1,svq3,\
theora,tiff,utvideo,v210,v210x,v308,v410,vc1,vc1image,vp3,vp5,vp6,vp6a,\
vp6f,vp7,vp8,vp9,webp,wmv1,wmv2,wmv3,wmv3image,y41p,ylc,yuv4,zlib"

../configure "${FFMPEG_BASE_OPTS[@]}" --prefix="$LOCALDESTDIR/opt/lightffmpeg" \
    --disable-{programs,devices,filters,encoders,muxers,debug,sdl2,network,protocols,doc,decoders} \
    --enable-decoder="${decoders}" --enable-gpl \
    --disable-bsf=aac_adtstoasc,text2movsub,noise,dca_core,mov2textsub,mp3_header_decompress \
    --disable-parser=aac,aac_latm,mpegaudio,tak,ac3,adx,dvaudio,opus,dvbsub,cavsvideo,dvd_nav,dvdsub,dca,flac
```

**Comment by wiiaboo** [comment]
To finally accept this, I'd want it behind a separate option, preferably, since most users might not expect a crippled x264. If you're not up to it, I can add it myself.

File: build/media-suite_compile.sh [hunk 1]
```diff
@@ -964,19 +964,30 @@ if [[ $x264 != no ]]; then
             --bindir="$LOCALDESTDIR/bin-video")
         old_PKG_CONFIG_PATH="$PKG_CONFIG_PATH"
         PKG_CONFIG_PATH="$LOCALDESTDIR/opt/lightffmpeg/lib/pkgconfig:$MINGW_PREFIX/lib/pkgconfig"
-        if [[ $standalone = y && $x264 = full ]]; then
+        if [[ $standalone = y && ( $x264 = full ||  $x264 = fullv ) ]]; then
```
Comment (wiiaboo) [inline build/media-suite_compile.sh]:
```bash
[[ $standalone = y ]] && [[ $x264 = full || $x264 = fullv ]]
```
or like below
```bash
[[ $standalone = y && $x264 =~ (full|fullv) ]]
```

File: media-autobuild_suite.bat [hunk 2]
```diff
@@ -357,6 +358,7 @@ if %buildx264%==2 set "x2642=no"
 if %buildx264%==3 set "x2642=high"
 if %buildx264%==4 set "x2642=full"
 if %buildx264%==5 set "x2642=shared"
+if %buildx264%==6 set "x2642=fullv"
 if %buildx264% GTR 5 GOTO x264
```
Comment (wiiaboo) [inline media-autobuild_suite.bat]: `GTR 6` or it will ignore value 6.

**Comment by wiiaboo** [comment]
Merged, thanks.

**Pull request closed without merging**